src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -1518,8 +1518,28 @@ private static String getFieldTypeByFieldName(String name) {
         String outerField = fieldsArray[OUTER_FIELD_INDEX];
         Gson gson = new Gson();
         HashMap fieldsMap = gson.fromJson(fields, HashMap.class);
-        HashMap propertiesMap = gson.fromJson(String.valueOf(fieldsMap.get(PROPERTIES)), HashMap.class);
-        return getFieldTypeByArrayLength(fieldsArray, outerField, gson, propertiesMap);
+        HashMap propertiesMap = getField(gson, fieldsMap, PROPERTIES);
+        HashMap finalField = getField(gson, propertiesMap, outerField);
+        for (String field : Arrays.copyOfRange(fieldsArray, INNER_FIELD_INDEX, fieldsArray.length)) {
```

Comment:
@gmakharat what do you think if I add an if statement here that checks the length of `fieldsArray`? 
it will be clearer to understand that if the length of `fieldsArray` is greater than one it means that we are looking for a nested field, otherwise we are looking for the outer field in the root and there's no need to go in the loop.

Technically it does not change anything but it will be easier to reason about it

Reply:
sounds good, you can use already existing method `fieldIsNested` I think, but I would rename it with `isFieldNested`.